CI: Use gci in the golangci-lint

Madhu-1 opened the pull request:
adding one more formater to check the import order as per the cephcsi documented way https://github.com/ceph/ceph-csi/blob/devel/docs/coding.md#imports  so that we dont get into this problem again.

https://golangci-lint.run/docs/formatters/configuration/#gci contains more details

Madhu-1 committed:
cleanup: fix the import as per the standard

Fixing the import order as per the cephcsi
documented standard.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: add gci for the formatting

adding the gci to check the import format
as per the cephcsi documented order.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

nixpanic commented on deploy/cephfs/kubernetes/csidriver.yaml:
```diff
@@ -11,7 +11,7 @@ kind: CSIDriver
   name: "cephfs.csi.ceph.com"
 spec:
-  attachRequired: true
+  attachRequired: false
```
how did this get included in the commit?

Madhu-1 replied:
This got added for `make generate-deploy` ~looks like we are not running this in CI~ this is not changed manually

nixpanic replied:
Oh, I thought there was a check that runs it and complains if there are uncommitted changed files :thinking:

Madhu-1 replied:
i checked the action not sure why this was not found

nixpanic commented on e2e/pvc.go:
```diff
@@ -408,7 +408,7 @@ func checkPVSelectorValuesForPVC(f *framework.Framework, pvc *v1.PersistentVolum
 
 func getMetricsForPVC(f *framework.Framework, pvc *v1.PersistentVolumeClaim, t int) error {
-	if (k8sBrokenMetrics(f.ClientSet)) {
+	if k8sBrokenMetrics(f.ClientSet) {
```
this is a different issue/fix... you could split it out, but that's up to you

Madhu-1 replied:
it was done by gci tool, lets keep it as it doesnt hurt

Madhu-1 committed:
cleanup: add attachRequired for cephfs

attachRequired is set to true for
cephFS driver, we missed adding it
to the generated template.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
@nixpanic there was a fix required in the api folder. PTAL found the problem still not sure why it was not caught in the CI

Madhu-1 commented:
@mergifyio queue
